corporate/lib/analytics.py

```diff
@@ -147,3 +153,60 @@ def get_plan_data_by_remote_realm() -> Dict[int, Dict[int, RemoteActivityPlanDat
                 current_server_data[realm_id] = plan_data
 
     return remote_server_plan_data_by_realm
+
+
+def get_realm_user_counts(
+    event_time: datetime = timezone_now(),
+) -> Dict[int, RemoteCustomerUserCount]:  # nocoverage
+    user_counts_by_realm: Dict[int, RemoteCustomerUserCount] = {}
+    for log in (
+        RemoteRealmAuditLog.objects.filter(
+            event_type__in=RemoteRealmAuditLog.SYNCED_BILLING_EVENTS,
+            event_time__lte=event_time,
+            remote_realm__isnull=False,
+        )
+        # Important: extra_data is empty for some pre-2020 audit logs
+        # prior to the introduction of realm_user_count_by_role
+        # logging. Meanwhile, modern Zulip servers using
+        # bulk_create_users to create the users in the system bot
+        # realm also generate such audit logs. Such audit logs should
+        # never be the latest in a normal realm.
+        .exclude(extra_data={})
+        .order_by("remote_realm", "-event_time")
+        .distinct("remote_realm")
+    ):
+        assert log.remote_realm is not None
+        user_counts_by_realm[log.remote_realm.id] = get_remote_customer_user_count([log])
+
+    return user_counts_by_realm
+
+
+def get_server_audit_logs(
+    event_time: datetime = timezone_now(),
+) -> Dict[int, List[RemoteRealmAuditLog]]:
+    logs = (
+        RemoteRealmAuditLog.objects.filter(
+            event_type__in=RemoteRealmAuditLog.SYNCED_BILLING_EVENTS,
+            event_time__lte=event_time,
+        )
```

Comment:
The query itself looks correct.

We probably want to add a database index on `RemoteRealmAuditLog` to make these queries more efficient. This is probably easiest done testing in production.

That said, if we want to make this fast, probably the right path involves caching the data -- we store it in `LicenseLedger` for paid plans; I guess we could potentially maintain that data structure even for the base plans.